contracts/DKVDaggerHashimoto.sol

```diff
@@ -277,32 +358,57 @@ contract DecentralizedKVDaggerHashimoto is DecentralizedKV {
         uint256 startShardId,
         uint256 shardLenBits,
         bytes32 hash0,
+        address miner,
         bytes32[][] memory proofsDim2,
         bytes[] memory maskedData
     ) internal view returns (bytes32) {
-        require(maskedData.length == randomChecks, "data vs checks: length mismatch");
-        require(proofsDim2.length == randomChecks, "proofs vs checks: length mismatch");
-        uint256 maxKvSize = 1 << maxKvSizeBits;
+        require(
+            maskedData.length == randomChecks,
+            "data vs checks: length mismatch"
+        );
+        require(
+            proofsDim2.length == randomChecks,
+            "proofs vs checks: length mismatch"
+        );
+        // calculate the number of chunks range of the sample check
         uint256 rows = 1 << (shardEntryBits + shardLenBits + chunkLenBits);
 
         for (uint256 i = 0; i < randomChecks; i++) {
-            require(maskedData[i].length == chunkSize, "invalid proof size");
+            uint256 mChunkSize = chunkSize;
```

Comment:
Minor: unused var. This assignment is not used by later logic

Reply:
the mChunkSize replaces the chunkSize
![截屏2023-04-01 12 37 18](https://user-images.githubusercontent.com/52317293/229265872-49311eec-0385-4ee3-860f-f191246bcfa3.png)

Using a uniform memory variable can reduce some gas consumption when it will use more times.

Reply:
I think it is too early for such optimization - especially since the gain is about 35 * 16 = 560 gas

Reply:
> the mChunkSize replaces the chunkSize ![截屏2023-04-01 12 37 18](https://user-images.githubusercontent.com/52317293/229265872-49311eec-0385-4ee3-860f-f191246bcfa3.png)
> 
> Using a uniform memory variable can reduce some gas consumption when it will use more times.

Oh sorry I am not aware of that. 
Meantime I feel a bit dangerous to do memory sharing without a proper disclaim/protection in structure-wise. 

As Qi said, might too early for now.